File: src/minigames/pillar_battle/src/main/java/work/lclpnet/ap2/game/pillar_battle/item/PotionItemClass.java
package work.lclpnet.ap2.game.pillar_battle.item;

import net.minecraft.core.component.DataComponents;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.alchemy.PotionContents;
import work.lclpnet.ap2.impl.util.ItemHelper;

import java.util.Random;
import java.util.stream.Stream;

public class PotionItemClass implements ItemClass {

    private final Item item;

    public PotionItemClass(Item item) {
        this.item = item;
    }

    @Override
    public ItemStack getRandomStack(Random random) {
        ItemStack stack = new ItemStack(item);

        var potion = ItemHelper.getRandomPotion(random);

        stack.set(DataComponents.POTION_CONTENTS, new PotionContents(potion));

        return stack;
    }

    @Override
    public Stream<Item> stream() {
        return Stream.of(item);
    }
}

File: src/minigames/pillar_battle/src/main/kotlin/work/lclpnet/ap2/game/pillar_battle/item/PotionItemClass.kt
package work.lclpnet.ap2.game.pillar_battle.item

import net.minecraft.core.component.DataComponents
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.alchemy.PotionContents
import work.lclpnet.ap2.impl.util.ItemHelper
import java.util.*
import java.util.stream.Stream

class PotionItemClass(private val item: Item) : ItemClass {
    override fun getRandomStack(random: Random): ItemStack {
        val stack = ItemStack(item)
        stack.set(DataComponents.POTION_CONTENTS, PotionContents(ItemHelper.getRandomPotion(random)))
        return stack
    }

    override fun stream(): Stream<Item> = Stream.of(item)
}
